tawazi/node/node.py

```diff
@@ -425,8 +426,9 @@ def __init__(
         """
         if isinstance(func, partial):
             # forward the qualname and annotations to the partial function
-            func.__qualname__ = func.func.__qualname__
-            func.__annotations__ = func.func.__annotations__
+            func = functools.update_wrapper(func, func.func)
+            # func.__qualname__ = func.func.__qualname__
+            # func.__annotations__ = func.func.__annotations__
```

Comment:
Of course! what were we thinking before!!

let's just remove the commented code below

Reply:
yep I just did !